pandora-client-web/src/components/settings/securitySettings.tsx

```diff
@@ -0,0 +1,273 @@
+			<ExtendCurrentSessionDialog token={ token } hide={ hide } />
+		</>
+	);
+}
+
+function ExtendCurrentSessionDialog({ token, hide }: { token: AuthToken; hide: () => boolean; }): ReactElement {
+	const directory = useDirectoryConnector();
+	const [password, setPassword] = React.useState('');
+	const now = useCurrentTime(60_000);
+
+	const [extend, processing] = useAsyncEvent(
+		() => directory.extendAuthToken(password),
+		(result) => {
+			if (result)
+				hide();
+			else
+				setPassword('');
+		},
+	);
+
+	return (
+		<ModalDialog>
+			<Form dirty={ false } onSubmit={ extend }>
+				<p>Your session will expire in { FormatTimeInterval(token.expires - now, 'short') }</p>
+				<FormField>
+					<label htmlFor='extend-current-session-password'>Password</label>
+					<input
+						type='password'
+						id='extend-current-session-password'
+						value={ password }
+						onChange={ (e) => setPassword(e.target.value) }
+					/>
+				</FormField>
+				<br />
+				<Row>
+					<Button onClick={ hide } disabled={ processing }>Cancel</Button>
+					<Button type='submit' disabled={ processing }>Extend</Button>
+				</Row>
+			</Form>
+		</ModalDialog>
+	);
+}
+
+interface PasswordChangeFormData {
+	oldPassword: string;
+	newPassword: string;
+	newPasswordConfirm: string;
+}
+
+function PasswordChange({ account }: { account: IDirectoryAccountInfo; }): ReactElement {
+	const directoryConnector = useDirectoryConnector();
+	const [invalidPassword, setInvalidPassword] = React.useState('');
+
+	const {
+		formState: { errors, submitCount, isSubmitting },
+		reset,
+		getValues,
+		handleSubmit,
+		register,
+		trigger,
+	} = useForm<PasswordChangeFormData>({ shouldUseNativeValidation: true, progressive: true });
+
+	React.useEffect(() => {
+		if (invalidPassword) {
+			void trigger();
+		}
+	}, [invalidPassword, trigger]);
+
+	const onSubmit = handleSubmit(async ({ oldPassword, newPassword }) => {
+		const passwordSha512Old = await PrehashPassword(oldPassword);
+		const passwordSha512New = await PrehashPassword(newPassword);
+		const { cryptoKey, onSuccess } = await directoryConnector.directMessageHandler.passwordChange(account.username, newPassword);
+
+		const resp = await directoryConnector.awaitResponse('passwordChange', {
+			passwordSha512Old,
+			passwordSha512New,
+			cryptoKey,
+		});
+
+		switch (resp.result) {
+			case 'ok':
+				onSuccess();
+				reset();
+				break;
+			case 'invalidPassword':
+				setInvalidPassword('Invalid password');
+				break;
+			default:
+				AssertNever(resp.result);
+		}
+	});
+
+	return (
+		<FieldsetToggle legend='Password Change' open={ false }>
+			<Form dirty={ submitCount > 0 } onSubmit={ onSubmit }>
+				<FormField>
+					<label htmlFor='password-change-old'>Old password</label>
+					<input
+						type='password'
+						id='password-change-old'
+						{ ...register('oldPassword', {
+							required: 'Old password is required',
+							validate: FormCreateStringValidator(PasswordSchema, 'password'),
```

Comment:
Current password shouldn't be validated - doing so would make it impossible to change password for anyone who's current password doesn't satisfy our password guidelines... and that is something we definitely don't want.
And yes, my testing account's doesn't satisfy this - that's how I noticed. This would be especially problematic if we ever changed the guidelines for how password should look like.